{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"gpuType":"A100"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"},"accelerator":"GPU"},"cells":[{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"hwis0iczQGYQ","executionInfo":{"status":"ok","timestamp":1746395172475,"user_tz":240,"elapsed":813,"user":{"displayName":"Xien Chen","userId":"14081978937262535291"}},"outputId":"8405a4d4-1df7-4b3d-9dde-4b5f87308346"},"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[31mERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'\u001b[0m\u001b[31m\n","\u001b[0m"]}],"source":[]},{"cell_type":"code","source":["import os\n","from google.colab import drive\n","from google.colab import auth\n","from google.auth import default\n","\n","drive.mount('/content/drive')\n","%cd /content/drive/MyDrive/workspace/normal_estimation"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"WnzYCyEhQjDf","executionInfo":{"status":"ok","timestamp":1746397767763,"user_tz":240,"elapsed":835,"user":{"displayName":"Xien Chen","userId":"14081978937262535291"}},"outputId":"0b7d254a-d42a-4e95-c6cb-00f5aeadf07c"},"execution_count":26,"outputs":[{"output_type":"stream","name":"stdout","text":["Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n","/content/drive/MyDrive/workspace/normal_estimation\n"]}]},{"cell_type":"code","source":["!pip install -r requirements.txt"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"id":"zH7KxyE3Q6Nx","executionInfo":{"status":"ok","timestamp":1746397772561,"user_tz":240,"elapsed":2413,"user":{"displayName":"Xien Chen","userId":"14081978937262535291"}},"outputId":"8f657864-cecf-46fb-a974-e836ecad7d2d"},"execution_count":27,"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: digit-interface>=0.2.1 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 1)) (0.2.1)\n","Requirement already satisfied: matplotlib>=3.5.3 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 2)) (3.10.0)\n","Requirement already satisfied: numpy>=1.23.2 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 3)) (2.0.2)\n","Requirement already satisfied: numpy-quaternion>=2022.4.2 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 4)) (2024.0.8)\n","Requirement already satisfied: opencv-python>=4.6.0.66 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 5)) (4.11.0.86)\n","Requirement already satisfied: pathtools>=0.1.2 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 6)) (0.1.2)\n","Requirement already satisfied: PyYAML>=6.0 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 7)) (6.0.2)\n","Requirement already satisfied: torch>=1.12.1 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 8)) (2.6.0+cu124)\n","Requirement already satisfied: torchvision>=0.13.1 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 9)) (0.21.0+cu124)\n","Requirement already satisfied: typing>=3.7.4.3 in /usr/local/lib/python3.11/dist-packages (from -r requirements.txt (line 10)) (3.7.4.3)\n","Requirement already satisfied: pyudev in /usr/local/lib/python3.11/dist-packages (from digit-interface>=0.2.1->-r requirements.txt (line 1)) (0.24.3)\n","Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.5.3->-r requirements.txt (line 2)) (1.3.2)\n","Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.5.3->-r requirements.txt (line 2)) (0.12.1)\n","Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.5.3->-r requirements.txt (line 2)) (4.57.0)\n","Requirement already satisfied: kiwisolver>=1.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.5.3->-r requirements.txt (line 2)) (1.4.8)\n","Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.5.3->-r requirements.txt (line 2)) (24.2)\n","Requirement already satisfied: pillow>=8 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.5.3->-r requirements.txt (line 2)) (11.2.1)\n","Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.5.3->-r requirements.txt (line 2)) (3.2.3)\n","Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.11/dist-packages (from matplotlib>=3.5.3->-r requirements.txt (line 2)) (2.9.0.post0)\n","Requirement already satisfied: scipy<2,>=1.5 in /usr/local/lib/python3.11/dist-packages (from numpy-quaternion>=2022.4.2->-r requirements.txt (line 4)) (1.15.2)\n","Requirement already satisfied: filelock in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (3.18.0)\n","Requirement already satisfied: typing-extensions>=4.10.0 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (4.13.2)\n","Requirement already satisfied: networkx in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (3.4.2)\n","Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (3.1.6)\n","Requirement already satisfied: fsspec in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (2025.3.2)\n","Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (12.4.127)\n","Requirement already satisfied: nvidia-cuda-runtime-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (12.4.127)\n","Requirement already satisfied: nvidia-cuda-cupti-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (12.4.127)\n","Requirement already satisfied: nvidia-cudnn-cu12==9.1.0.70 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (9.1.0.70)\n","Requirement already satisfied: nvidia-cublas-cu12==12.4.5.8 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (12.4.5.8)\n","Requirement already satisfied: nvidia-cufft-cu12==11.2.1.3 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (11.2.1.3)\n","Requirement already satisfied: nvidia-curand-cu12==10.3.5.147 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (10.3.5.147)\n","Requirement already satisfied: nvidia-cusolver-cu12==11.6.1.9 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (11.6.1.9)\n","Requirement already satisfied: nvidia-cusparse-cu12==12.3.1.170 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (12.3.1.170)\n","Requirement already satisfied: nvidia-cusparselt-cu12==0.6.2 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (0.6.2)\n","Requirement already satisfied: nvidia-nccl-cu12==2.21.5 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (2.21.5)\n","Requirement already satisfied: nvidia-nvtx-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (12.4.127)\n","Requirement already satisfied: nvidia-nvjitlink-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (12.4.127)\n","Requirement already satisfied: triton==3.2.0 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (3.2.0)\n","Requirement already satisfied: sympy==1.13.1 in /usr/local/lib/python3.11/dist-packages (from torch>=1.12.1->-r requirements.txt (line 8)) (1.13.1)\n","Requirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.11/dist-packages (from sympy==1.13.1->torch>=1.12.1->-r requirements.txt (line 8)) (1.3.0)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.7->matplotlib>=3.5.3->-r requirements.txt (line 2)) (1.17.0)\n","Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from jinja2->torch>=1.12.1->-r requirements.txt (line 8)) (3.0.2)\n"]}]},{"cell_type":"code","source":["!mkdir -p datasets"],"metadata":{"id":"eErO9Et7VSM9"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["!pip install --upgrade huggingface_hub"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"NijF8MzCVJcU","executionInfo":{"status":"ok","timestamp":1746398795649,"user_tz":240,"elapsed":2331,"user":{"displayName":"Xien Chen","userId":"14081978937262535291"}},"outputId":"162b6e37-31d9-459a-8a12-4d8d9388f359","collapsed":true},"execution_count":32,"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: huggingface_hub in /usr/local/lib/python3.11/dist-packages (0.30.2)\n","Requirement already satisfied: filelock in /usr/local/lib/python3.11/dist-packages (from huggingface_hub) (3.18.0)\n","Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.11/dist-packages (from huggingface_hub) (2025.3.2)\n","Requirement already satisfied: packaging>=20.9 in /usr/local/lib/python3.11/dist-packages (from huggingface_hub) (24.2)\n","Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.11/dist-packages (from huggingface_hub) (6.0.2)\n","Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from huggingface_hub) (2.32.3)\n","Requirement already satisfied: tqdm>=4.42.1 in /usr/local/lib/python3.11/dist-packages (from huggingface_hub) (4.67.1)\n","Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.11/dist-packages (from huggingface_hub) (4.13.2)\n","Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface_hub) (3.4.1)\n","Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface_hub) (3.10)\n","Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface_hub) (2.4.0)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface_hub) (2025.4.26)\n"]}]},{"cell_type":"code","source":["!huggingface-cli login"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"RU91hSQfewPK","executionInfo":{"status":"ok","timestamp":1746398865114,"user_tz":240,"elapsed":55112,"user":{"displayName":"Xien Chen","userId":"14081978937262535291"}},"outputId":"bd7c5976-b4ac-4071-ed78-447b3c71690b"},"execution_count":33,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","    _|    _|  _|    _|    _|_|_|    _|_|_|  _|_|_|  _|      _|    _|_|_|      _|_|_|_|    _|_|      _|_|_|  _|_|_|_|\n","    _|    _|  _|    _|  _|        _|          _|    _|_|    _|  _|            _|        _|    _|  _|        _|\n","    _|_|_|_|  _|    _|  _|  _|_|  _|  _|_|    _|    _|  _|  _|  _|  _|_|      _|_|_|    _|_|_|_|  _|        _|_|_|\n","    _|    _|  _|    _|  _|    _|  _|    _|    _|    _|    _|_|  _|    _|      _|        _|    _|  _|        _|\n","    _|    _|    _|_|      _|_|_|    _|_|_|  _|_|_|  _|      _|    _|_|_|      _|        _|    _|    _|_|_|  _|_|_|_|\n","\n","    To log in, `huggingface_hub` requires a token generated from https://huggingface.co/settings/tokens .\n","Enter your token (input will not be visible): \n","Add token as git credential? (Y/n) n\n","Token is valid (permission: fineGrained).\n","The token `dfd` has been saved to /root/.cache/huggingface/stored_tokens\n","Your token has been saved to /root/.cache/huggingface/token\n","Login successful.\n","The current active token is: `dfd`\n"]}]},{"cell_type":"code","source":[],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"id":"1hxMJC0MfBST","executionInfo":{"status":"ok","timestamp":1746398880712,"user_tz":240,"elapsed":727,"user":{"displayName":"Xien Chen","userId":"14081978937262535291"}},"outputId":"dd77e2aa-bb49-4995-e6bf-955f84d6bef8"},"execution_count":34,"outputs":[{"output_type":"stream","name":"stdout","text":["Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/utils/_http.py\", line 409, in hf_raise_for_status\n","    response.raise_for_status()\n","  File \"/usr/local/lib/python3.11/dist-packages/requests/models.py\", line 1024, in raise_for_status\n","    raise HTTPError(http_error_msg, response=self)\n","requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://huggingface.co/api/models/facebook/gelsight-force-estimation/revision/main\n","\n","The above exception was the direct cause of the following exception:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/bin/huggingface-cli\", line 10, in <module>\n","    sys.exit(main())\n","             ^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/commands/huggingface_cli.py\", line 57, in main\n","    service.run()\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/commands/download.py\", line 153, in run\n","    print(self._download())  # Print path to downloaded files\n","          ^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/commands/download.py\", line 187, in _download\n","    return snapshot_download(\n","           ^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/utils/_validators.py\", line 114, in _inner_fn\n","    return fn(*args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/_snapshot_download.py\", line 232, in snapshot_download\n","    raise api_call_error\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/_snapshot_download.py\", line 155, in snapshot_download\n","    repo_info = api.repo_info(repo_id=repo_id, repo_type=repo_type, revision=revision, token=token)\n","                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/utils/_validators.py\", line 114, in _inner_fn\n","    return fn(*args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/hf_api.py\", line 2807, in repo_info\n","    return method(\n","           ^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/utils/_validators.py\", line 114, in _inner_fn\n","    return fn(*args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/hf_api.py\", line 2592, in model_info\n","    hf_raise_for_status(r)\n","  File \"/usr/local/lib/python3.11/dist-packages/huggingface_hub/utils/_http.py\", line 459, in hf_raise_for_status\n","    raise _format(RepositoryNotFoundError, message, response) from e\n","huggingface_hub.errors.RepositoryNotFoundError: 404 Client Error. (Request ID: Root=1-6817eea0-5d2cbf1572b67c6842a7d831;7371a46d-858b-4cf1-8442-ec9ad1d21ba4)\n","\n","Repository Not Found for url: https://huggingface.co/api/models/facebook/gelsight-force-estimation/revision/main.\n","Please make sure you specified the correct `repo_id` and `repo_type`.\n","If you are trying to access a private or gated repo, make sure you are authenticated. For more details, see https://huggingface.co/docs/huggingface_hub/authentication\n"]}]},{"cell_type":"code","source":["!pip install rospkg catkin_pkg defusedxml"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"collapsed":true,"id":"I4L966LCVVYX","executionInfo":{"status":"ok","timestamp":1746397782493,"user_tz":240,"elapsed":2212,"user":{"displayName":"Xien Chen","userId":"14081978937262535291"}},"outputId":"73db8bcd-3777-4abd-871d-8bb42f087338"},"execution_count":28,"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: rospkg in /usr/local/lib/python3.11/dist-packages (1.6.0)\n","Requirement already satisfied: catkin_pkg in /usr/local/lib/python3.11/dist-packages (1.0.0)\n","Requirement already satisfied: defusedxml in /usr/local/lib/python3.11/dist-packages (0.7.1)\n","Requirement already satisfied: PyYAML in /usr/local/lib/python3.11/dist-packages (from rospkg) (6.0.2)\n","Requirement already satisfied: distro>=1.4.0 in /usr/local/lib/python3.11/dist-packages (from rospkg) (1.9.0)\n","Requirement already satisfied: docutils in /usr/local/lib/python3.11/dist-packages (from catkin_pkg) (0.21.2)\n","Requirement already satisfied: python-dateutil in /usr/local/lib/python3.11/dist-packages (from catkin_pkg) (2.9.0.post0)\n","Requirement already satisfied: pyparsing in /usr/local/lib/python3.11/dist-packages (from catkin_pkg) (3.2.3)\n","Requirement already satisfied: setuptools in /usr/local/lib/python3.11/dist-packages (from catkin_pkg) (75.2.0)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil->catkin_pkg) (1.17.0)\n"]}]},{"cell_type":"code","source":["!python src/canfnet/scripts/train/train.py \\\n","  --device cuda \\\n","  --epochs 10 --batch-size 32 \\\n","  --load models/DIGIT/model_24-02-2023_14-49-02_256_digit.pth \\\n","  --old-ckpt models/DIGIT/model_24-02-2023_14-49-02_256_digit.pth \\\n","  --lwf-lambda 1.0 \\\n","  --force-max 3.0"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"Wt12uMs7TtCs","executionInfo":{"status":"ok","timestamp":1746395976136,"user_tz":240,"elapsed":11330,"user":{"displayName":"Xien Chen","userId":"14081978937262535291"}},"outputId":"5540b07f-e5ed-4b80-bfb3-db705bcc4548"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Traceback (most recent call last):\n","  File \"/content/drive/MyDrive/workspace/normal_estimation/src/canfnet/scripts/train/train.py\", line 24, in <module>\n","    import neptune.new as neptune\n","ModuleNotFoundError: No module named 'neptune'\n"]}]}]}